lib/semantics/expression.cc

```diff
@@ -1515,18 +1564,31 @@ std::optional<ActualArgument> ExpressionAnalyzer::AnalyzeActualArgument(
   if (const Symbol * assumedTypeDummy{AssumedTypeDummy(expr)}) {
     return ActualArgument{ActualArgument::AssumedType{*assumedTypeDummy}};
   } else if (MaybeExpr argExpr{Analyze(expr)}) {
-    return ActualArgument{Fold(GetFoldingContext(), std::move(*argExpr))};
-  } else {
-    return std::nullopt;
-  }
-}
-
-std::optional<ActualArgument> ExpressionAnalyzer::AnalyzeActualArgument(
-    const parser::Variable &var) {
-  if (const Symbol * assumedTypeDummy{AssumedTypeDummy(var)}) {
-    return ActualArgument{ActualArgument::AssumedType{*assumedTypeDummy}};
-  } else if (MaybeExpr argExpr{Analyze(var)}) {
-    return ActualArgument{std::move(*argExpr)};
+    Expr<SomeType> x{Fold(GetFoldingContext(), std::move(*argExpr))};
+    if (const auto *proc{std::get_if<ProcedureDesignator>(&x.u)}) {
+      if (!std::holds_alternative<SpecificIntrinsic>(proc->u) &&
+          proc->IsElemental()) {  // C1533
+        Say(expr.source,
+            "Non-intrinsic ELEMENTAL procedure cannot be passed as argument."_err_en_US);
```

Comment:
We haven't been putting periods at the end of error messages.

Reply:
Ok.